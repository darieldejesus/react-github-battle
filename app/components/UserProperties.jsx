import React from 'react';
import PropTypes from 'prop-types';

/**
 * This is the component to show the profile info.
 */
const UserProperties = ({ profile }) => (
  <ul className='space-list-items'>
    {profile.name && <li>{profile.name}</li>}
    {profile.location && <li>{profile.location}</li>}
    {profile.company && <li>{profile.company}</li>}
    <li><b>Followers:</b> {profile.followers}</li>
    <li><b>Following:</b> {profile.following}</li>
    <li><b>Public Repos:</b> {profile.public_repos}</li>
    {profile.blog && <li><a href={profile.blog}>{profile.blog}</a></li>}
  </ul>
)

UserProperties.propTypes = {
  profile: PropTypes.shape({
    followers: PropTypes.number.isRequired,
    following: PropTypes.number.isRequired,
    public_repos: PropTypes.number.isRequired,
    blog: PropTypes.string,
    company: PropTypes.string,
    location: PropTypes.string,
    name: PropTypes.string
  })
}

export default UserProperties;
