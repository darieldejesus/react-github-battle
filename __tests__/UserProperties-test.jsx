import React from 'react';
import { shallow } from 'enzyme';

import UserProperties from '../app/components/UserProperties';

describe('<UserProperties />', () => {
  it('User Properties is rendered with the required props', () => {
    const profile = {
      login: 'username',
      avatar_url: 'https://example.com/username.jpg',
      followers: 10,
      following: 5,
      public_repos: 3
    };
    const userProperties = shallow(
      <UserProperties profile={profile} />
    );
    expect(userProperties).toMatchSnapshot();
  });
});