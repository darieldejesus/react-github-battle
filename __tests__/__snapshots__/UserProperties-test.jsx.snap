// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserProperties /> User Properties is rendered with the required props 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <ul
    className="space-list-items"
  >
    <li>
      <b>
        Followers:
      </b>
       
      10
    </li>
    <li>
      <b>
        Following:
      </b>
       
      5
    </li>
    <li>
      <b>
        Public Repos:
      </b>
       
      3
    </li>
  </ul>,
  "nodes": Array [
    <ul
      className="space-list-items"
    >
      <li>
        <b>
          Followers:
        </b>
         
        10
      </li>
      <li>
        <b>
          Following:
        </b>
         
        5
      </li>
      <li>
        <b>
          Public Repos:
        </b>
         
        3
      </li>
    </ul>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <UserProperties
        profile={
          Object {
            "avatar_url": "https://example.com/username.jpg",
            "followers": 10,
            "following": 5,
            "login": "username",
            "public_repos": 3,
          }
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "profile": Object {
            "avatar_url": "https://example.com/username.jpg",
            "followers": 10,
            "following": 5,
            "login": "username",
            "public_repos": 3,
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <ul
          className="space-list-items"
        >
          <li>
            <b>
              Followers:
            </b>
             
            10
          </li>
          <li>
            <b>
              Following:
            </b>
             
            5
          </li>
          <li>
            <b>
              Public Repos:
            </b>
             
            3
          </li>
        </ul>,
        "_debugID": 2,
        "_renderedOutput": <ul
          className="space-list-items"
        >
          <li>
            <b>
              Followers:
            </b>
             
            10
          </li>
          <li>
            <b>
              Following:
            </b>
             
            5
          </li>
          <li>
            <b>
              Public Repos:
            </b>
             
            3
          </li>
        </ul>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <UserProperties
    profile={
      Object {
        "avatar_url": "https://example.com/username.jpg",
        "followers": 10,
        "following": 5,
        "login": "username",
        "public_repos": 3,
      }
    }
  />,
}
`;
